### ISSUE TITLE
[HOLD for payment 2021-11-04] Chat - Chronos message is cut off at the edge and not fully visible

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#5008]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Lunch the app
2. Log in with expensifail account
3. On Search box - find Chronos
4. Text - Top

## Expected Result:
Chronos messages are fully visible.
## Actual Result:
Chronos message isn't fully visible, it is cut off at the end
## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- iOS
- Android
- Mobile Web

**Version Number:**
1.0.91-0
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
![Bug5219680_Screenshot_20210901-160649_New_Expensify](https://user-images.githubusercontent.com/43996225/131765841-a61559ca-6e66-464a-a621-c4279e140570.jpg)

**Expensify/Expensify Issue URL:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @francoisl (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Christinadobrzyn (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by Christinadobrzyn** [comment]
External job created in Upwork - https://www.upwork.com/jobs/~0135dabda6dbb82f78
Internal job created in Upwork - https://www.upwork.com/ab/applicants/1433592402430713856/job-details

Feel free to submit a proposal @kavimuru!

**Comment by MelvinBot** [comment]
Triggered auto assignment to @johnmlee101 (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by equationalapplications** [comment]
I launched the web app in my development environment and logged in with my email. I clicked the search icon, but I can't find Chronos.

**Comment by pROFESOR11** [comment]
Hi @kavimuru ,
What exactly is `expensifail` account and how can I login with that?
I guess I need to login with that specific account to be able to reproduce the issue.

**Comment by Christinadobrzyn** [comment]
Hi @pROFESOR11! Sorry for the delayed response, Expensifail is our test account. You can create a test account to our NewDot product following the setup details in this [article](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md). 

@equationalapplications - Chronos is a 'contact' option in the NewDot chat. To start a conversation with Chronos, click the green + icon > New Chat >  type Chronos. 

![2021-09-10_10-36-20 (1)](https://user-images.githubusercontent.com/51066321/132790291-9ec04f32-daf4-4b72-badf-80d79a35928f.gif)

**Comment by Christinadobrzyn** [comment]
Doubled price to $500

**Comment by pROFESOR11** [comment]
@Christinadobrzyn Thanks for your reply. Well, I want to work the issue, but I am not sure what I am doing wrong, for some reason I can't reproduce the issue. I can't find `Chronos` in contacts. I tried to manually add `chronos@expensify.com` and wrote "Hey there!", but got no response. Should I write something specific to get an answer as in the issue description?

https://user-images.githubusercontent.com/26159043/133054400-305c0ee9-caab-48e4-9afe-add75f3d8b56.mov

**Comment by parasharrajat** [comment]
This issue is seemed to be fixed now. Please re-test @isagoico .
![Screenshot_1631541715](https://user-images.githubusercontent.com/24370807/133097741-3b11804a-0b55-4571-821b-47b24a60f98a.png)

**Comment by equationalapplications** [comment]
> Hi @pROFESOR11! Sorry for the delayed response, Expensifail is our test account. You can create a test account to our NewDot product following the setup details in this [article](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md).
> 
> @equationalapplications - Chronos is a 'contact' option in the NewDot chat. To start a conversation with Chronos, click the green + icon > New Chat > type Chronos.
> 
> ![2021-09-10_10-36-20 (1)](https://user-images.githubusercontent.com/51066321/132790291-9ec04f32-daf4-4b72-badf-80d79a35928f.gif)

@Christinadobrzyn thank you. I too cannot find Chronos after starting a new chat and searching for Chronos.

**Comment by Christinadobrzyn** [comment]
Thanks for confirming @equationalapplications! 

Based on what @parasharrajat mentioned above, we'll have @isagoico test this to see if it's fixed.

**Comment by parasharrajat** [comment]
@isagoico Please retest.

**Comment by Christinadobrzyn** [comment]
Pinged Isa in Slack - https://expensify.slack.com/archives/C01GTK53T8Q/p1631759290376300?thread_ts=1630630175.357600&cid=C01GTK53T8Q

**Comment by isagoico** [comment]
We were able to reproduce the issue in latest build 

https://user-images.githubusercontent.com/44479856/133655831-9bfce80b-3e40-48b1-84d2-b610b4b7fd38.mp4

**Comment by parasharrajat** [comment]
Good Thanks @isagoico, I will look closely. Maybe come up with follow up questions.

**Comment by Christinadobrzyn** [comment]
Looks like we're still waiting on proposals for this! Doubled the price to $1000

**Comment by equationalapplications** [comment]
I am not able to find Chronos in the chat search still running on my localhost

https://user-images.githubusercontent.com/65428263/134015518-8f149413-ea00-4fd9-8209-111c4a4a99c4.mp4

.

**Comment by equationalapplications** [comment]
Also, I cannot add a test account from the development web server on localhost

https://user-images.githubusercontent.com/65428263/134016241-e3f950fc-6354-490f-bb85-24eaba73005f.mp4

.

**Comment by equationalapplications** [comment]
Also, I cannot find Chronos in new.expensify.com in the browser. 
![Screenshot_20210920-115948](https://user-images.githubusercontent.com/65428263/134035097-bd3b7793-1948-471e-a8d4-32a43c777bad.png)

**Comment by parasharrajat** [comment]
Finally, the solution is found.

### Proposal
1. Li items are extending beyond the parent width and trying `flexShrink:1` which was suggested from Lib was not working as the container was allowing unrestricted width.

To fix this we have to restrict the container's width and then respectively `ul` tags.

We have to add styles for **ul & ol** in 
https://github.com/Expensify/App/blob/0a3da2e0bebf5105ea60a36cc453d6a1a9c312bf/src/styles/styles.js#L2036
```
      ul: {
            maxWidth: '100%',
            flex: 1, //Optional not needed but seems useful to me.
        },
 ```       
Then we apply `maxWidth: '100%'` to https://github.com/Expensify/App/blob/0a3da2e0bebf5105ea60a36cc453d6a1a9c312bf/src/styles/styles.js#L1051

which fixes this issue.

|BEFORE| AFTER|
|---------|----------|
| ![issue](https://user-images.githubusercontent.com/24370807/134061847-c3cb4c07-555f-47cd-bfac-38fecc8dad3d.png) | ![fix](https://user-images.githubusercontent.com/24370807/134061836-f4660912-3fb5-413c-b19f-ebe3b771af7f.png) |

**Comment by johnmlee101** [comment]
Looks good to me @parasharrajat !

**Comment by johnmlee101** [comment]
cc @Christinadobrzyn

**Comment by Christinadobrzyn** [comment]
Awesome! Hired @parasharrajat in Upwork.

**Comment by Christinadobrzyn** [comment]
still on hold for n6

**Comment by Christinadobrzyn** [comment]
N6 hold is removed as the newsletter has deployed. 

Here's the [PR](https://github.com/Expensify/App/pull/5364) for this job - a $250 bonus will also apply to this job!

**Comment by Christinadobrzyn** [comment]
7-day production window is upon us. Paid @parasharrajat in Upwork and included a $250 bonus. Closing this. 🎉
